### PR TITLE
Allow INSERT statement for Cassandra table having hidden id column

### PR DESCRIPTION
**Pull request by ebyhr**

### CODE REVIEW

File: presto-cassandra/src/test/java/io/prestosql/plugin/cassandra/TestCassandraIntegrationSmokeTest.java [hunk 1]
```diff
@@ -165,6 +165,16 @@ public void testSelect()
         assertSelect(TABLE_ALL_TYPES_PARTITION_KEY, false);
     }
 
+    @Test
+    public void testCreateTable()
```
Comment (losipiuk) [inline presto-cassandra/src/test/java/io/prestosql/plugin/cassandra/TestCassandraIntegrationSmokeTest.java:169]: Can you please name this test in a way that captures what is the functionality tested here. IMO `testInsertToTableWithHiddenId` would be better name.

File: presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraMetadata.java [hunk 2]
```diff
@@ -335,19 +335,32 @@ public ConnectorInsertTableHandle beginInsert(ConnectorSession session, Connecto
...
 
     @Override
     public Optional<ConnectorOutputMetadata> finishInsert(ConnectorSession session, ConnectorInsertTableHandle insertHandle, Collection<Slice> fragments, Collection<ComputedStatistics> computedStatistics)
     {
         return Optional.empty();
     }
+
+    private static boolean isHiddenIdColumn(CassandraColumnHandle columnHandle)
+    {
+        return columnHandle.isHidden() && "id".equals(columnHandle.getName());
```
Comment (losipiuk) [inline presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraMetadata.java:364]: "id" literal appears in couple of places already. Can you please extract constant for that.

File: presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraPageSinkProvider.java [hunk 3]
```diff
@@ -67,6 +67,6 @@ public ConnectorPageSink createPageSink(ConnectorTransactionHandle transactionHa
                 handle.getTableName(),
                 handle.getColumnNames(),
                 handle.getColumnTypes(),
-                false);
+                handle.isGenerateUuid());
```
Comment (losipiuk) [inline presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraPageSinkProvider.java:70]: Not sure what is proper casing according to code-style but `CassandraPageSing` uses `generateUUID`. Can you please unify?

Comment (ebyhr) [reply presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraPageSinkProvider.java:70]: Unified to `generateUuid` as we have several `Uuid~` classes.

File: presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraMetadata.java [hunk 4]
```diff
@@ -336,19 +336,32 @@ public ConnectorInsertTableHandle beginInsert(ConnectorSession session, Connecto
...
-        List<Type> columnTypes = columns.stream().map(CassandraColumnHandle::getType).collect(Collectors.toList());
+        List<String> columnNames = columns.stream()
+                .filter(columnHandle -> !isHiddenIdColumn(columnHandle))
+                .map(CassandraColumnHandle::getName)
+                .collect(Collectors.toList());
+        List<Type> columnTypes = columns.stream()
+                .filter(columnHandle -> !isHiddenIdColumn(columnHandle))
+                .map(CassandraColumnHandle::getType)
+                .collect(Collectors.toList());
+        boolean generateUuid = columns.stream().anyMatch(CassandraMetadata::isHiddenIdColumn);
```
Comment (findepi) [inline presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraMetadata.java:347]: Can there be more than one?

Comment (ebyhr) [reply presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraMetadata.java:347]: I suppose it's always one.

Comment (findepi) [reply presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraMetadata.java:347]: I guess (and it should), so the code here is too flexible

maybe
```
boolean generateUuid = columns.stream()
  .filter(CassandraMetadata::isHiddenIdColumn))
 .collect(toOptional()) // must be at most one
 .isPresent();
```

**Pull request merged**